src/charm.py

```diff
@@ -526,27 +529,25 @@ def _update_ca_certs(self):
             ca_cert_path.unlink(missing_ok=True)
         subprocess.run(["update-ca-certificates", "--fresh"], check=True)
 
-    def _get_peer_addresses(self) -> List[str]:
-        """Create a list of HA addresses of all peer units (all units excluding current).
+    def _get_peer_addresses(self, include_this_unit=True) -> List[str]:
+        """Returns a list of the addresses the peer units, without scheme or port.
```

Comment:
This is a good opportunity to start using correct terminology.

"address without scheme or port" is "hostname".

```mermaid
graph LR

scheme ---|://| username

subgraph netloc
username ---|:| password ---|@| hostname ---|:| port
end

port --- path["/path"]
```

![image](https://github.com/canonical/alertmanager-k8s-operator/assets/82407168/37e12565-8538-488e-8c76-3bf535ba842f)

Reply:
Agreed.  [this commit](https://github.com/canonical/alertmanager-k8s-operator/pull/264/commits/8efdae9454016a51312dbe21babaf96ffa94aa54) I think covers the inconsistencies.  It reaches into the WorkloadManager as well